### PR TITLE
feature/TIN-14

### PR DESCRIPTION
**Pull request by tsenovi**
JWT Token

### CODE REVIEW

File: src/main/java/com/volasoftware/tinder/configs/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  @Override
+  protected void doFilterInternal(
+      @NonNull HttpServletRequest request,
+      @NonNull HttpServletResponse response,
+      @NonNull FilterChain filterChain
+  ) throws ServletException, IOException {
+    final String authHeader = request.getHeader("Authorization");
+    final String jwt;
+    final String accountEmail;
+    if (authHeader == null || !authHeader.startsWith("Bearer ")) {
```
Comment (Kr1s4an) [inline src/main/java/com/volasoftware/tinder/configs/JwtAuthenticationFilter.java]: Extract Bearer into a constant.

File: src/main/java/com/volasoftware/tinder/configs/ApplicationConfig.java [hunk 2]
```diff
@@ -21,4 +27,23 @@ public UserDetailsService userDetailsService() {
...
+  public AuthenticationManager authenticationManager(AuthenticationConfiguration config)
+      throws Exception {
+    return config.getAuthenticationManager();
+  }
+
+  @Bean
+  public PasswordEncoder passwordEncoder() {
+    return new BCryptPasswordEncoder();
+  }
+
```
Comment (Kr1s4an) [inline src/main/java/com/volasoftware/tinder/configs/ApplicationConfig.java]: Remove unnecessary empty spaces.

File: src/main/java/com/volasoftware/tinder/models/Role.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.volasoftware.tinder.models;
+
+public enum Role {
+
+  USER,
+  Admin
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/models/Role.java]: Check the enum naming: https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html

File: src/test/java/com/volasoftware/tinder/controllers/AuthenticationControllerTest.java [hunk 4]
```diff
@@ -1,42 +1,49 @@
...
 
   @Autowired
   private MockMvc mockMvc;
 
+  @Autowired
   @MockBean
   private AccountService accountService;
 
+
+//  TODO fix the test method - Consider defining a bean of type 'javax.sql.DataSource' in your configuration.
```
Comment (todornikolov1) [inline src/test/java/com/volasoftware/tinder/controllers/AuthenticationControllerTest.java]: Is the method fine now?

Comment (tsenovi) [reply src/test/java/com/volasoftware/tinder/controllers/AuthenticationControllerTest.java]: Still needs to be fixed... Working on it!

File: src/main/java/com/volasoftware/tinder/models/Role.java (hunk 3 above)
Comment (tsenovi) [reply src/main/java/com/volasoftware/tinder/models/Role.java]: Admin was fixed by ADMIN

Comment (todornikolov1) [reply src/main/java/com/volasoftware/tinder/models/Role.java]: Thanks!

File: src/test/java/com/volasoftware/tinder/controllers/AuthenticationControllerTest.java (hunk 4 above)
Comment (tsenovi) [reply src/test/java/com/volasoftware/tinder/controllers/AuthenticationControllerTest.java]: Test was fixed

**Review by todornikolov1 — APPROVED** [review]

**Pull request merged**